decisiontree/views/views.py

```diff
@@ -0,0 +1,290 @@
+    template_name = "tree/path_list.html"
+
+
+class PathCreateUpdate(base.TreeCreateUpdateView):
+    create_success_message = "You have successfully inserted Question Path {obj.id}"
+    edit_success_message = "Path successfully updated"
+    form_class = forms.PathForm
+    model = models.Transition
+    success_url = reverse_lazy("path_list")
+    template_name = "tree/path.html"
+
+
+class PathDelete(base.TreeDeleteView):
+    model = models.Transition
+    success_message = "Path successfully deleted"
+    success_url = reverse_lazy("path_list")
+
+
+class QuestionList(base.TreeListView):
+    model = models.Question
+    order_by = ['text']
+    template_name = 'tree/questions_list.html'
+
+
+class QuestionCreateUpdate(base.TreeCreateUpdateView):
+    create_success_message = "You have successfully inserted a Question {obj.text}"
+    edit_success_message = "You have successfully updated the Question"
+    form_class = forms.QuestionForm
+    model = models.Question
+    success_url = reverse_lazy('list-questions')
+    template_name = 'tree/question.html'
+
+
+class QuestionDelete(base.TreeDeleteView):
+    model = models.Question
+    success_message = "Question successfully deleted"
+    success_url = reverse_lazy('list-questions')
+
+
+class StateList(base.TreeListView):
+    model = models.TreeState
+    order_by = ['question']
+    select_related = ['question']
+    template_name = "tree/states_list.html"
+
+
+class StateCreateUpdate(base.TreeCreateUpdateView):
+    create_success_message = "You have successfully inserted State {obj.name}."
+    edit_success_message = "State updated successfully"
+    model = models.TreeState
+    form_class = forms.StateForm
+    success_url = reverse_lazy("state_list")
+    template_name = "tree/state.html"
+
+
+class StateDelete(base.TreeDeleteView):
+    model = models.TreeState
+    success_message = "State successfully deleted"
+    success_url = reverse_lazy("state_list")
+
+
+class SurveyList(base.TreeListView):
+    model = models.Tree
+    order_by = ['trigger']
+    select_related = ['root_state__question']
+    template_name = 'tree/index.html'
+
+    def get_queryset(self):
+        return super(SurveyList, self).get_queryset().annotate(count=Count('sessions'))
+
+
+class SurveyExport(base.TreeDetailView):
+    model = models.Tree
+
+    def get(self, request, *args, **kwargs):
+        tree = self.get_object()
+        if tree.has_loops():
+            return redirect('list-surveys')
+        all_states = tree.get_all_states()
+        output = StringIO()
+        w = csv.writer(output)
+        headings = ["Person", "Date"]
+        headings.extend([state.question for state in all_states])
+        w.writerow(headings)
+        sessions = models.Session.objects.all().filter(tree=tree)
+        for session in sessions:
+            values = [str(session.connection), session.start_date]
+            transitions = map((lambda x: x.transition), session.entries.all())
```

Comment:
Hmm, I don't think that would work - looks like the code uses the transition object, but `.values_list()` would return the primary key. 

This view didn't fit the basic logic that all the other ones did, so I didn't refactor it. I hope to be able to do so as we continue to update the app.